pkg/apis/allocation/v1/gameserverallocation.go

```diff
@@ -607,7 +607,14 @@ type GameServerAllocationStatus struct {
 	NodeName       string                          `json:"nodeName,omitempty"`
 	// If the allocation is from a remote cluster, Source is the endpoint of the remote agones-allocator.
 	// Otherwise, Source is "local"
-	Source string `json:"source"`
+	Source   string              `json:"source"`
+	Metadata *GameServerMetadata `json:"metadata,omitempty"`
+}
+
+// GameServerMetadata is the metadata from the allocated game server at allocation time
+type GameServerMetadata struct {
```

Comment:
🤔  this is exactly the same at `MetaPatch` - I know the naming is not great, but should we just reuse the type?